components/layout/LogoutButton.js

```diff
@@ -1,11 +1,44 @@
 import Link from "next/link";
+import { useState } from "react";
+import {
+  getAuth,
+  signOut,
+  onAuthStateChanged
+} from "firebase/auth"
+import { firebaseApp } from "../../config/firebaseConfig";
 
-const LogoutButton = () => (
-  <Link href='/'>
-    <a>
-      <button type='button'>Logout</button>
-    </a>
-  </Link>
-);
 
-export default LogoutButton;
+const LogoutButton = () => {
+
+  const auth = getAuth(firebaseApp);
+  const [logoutNotification, setLogoutNotification] = useState('');
+
+  const handleLogout = (e) => {
+    e.preventDefault();
+    signOut(auth)
+      .then(() => {
+        console.log('User is logged out')
```

Comment:
This replaces the logoutNotification state that otherwise would have been used to indicate the user log-out. But have you thought of what this `.then()` promise is doing? It gives the console log only when the user log-out went without incident. But because you don't have a `.catch()` error handling following this, there'd be no log or any indication that the user's not logged out when the the app is routed by the Next.js <Link> back to the home page, which happens as long as the button is pressed regardless of the state of Firebase auth.

Reply:
Left just signOut functionality